### PR TITLE
Use the streamlined ToolchainCluster and update its CRD.

### PR DESCRIPTION
**Pull request by metlos**
Related PRs:
- api: https://github.com/codeready-toolchain/api/pull/448
- host-operator: https://github.com/codeready-toolchain/host-operator/pull/1094
- member-operator: https://github.com/codeready-toolchain/member-operator/pull/602
- registration-service: https://github.com/codeready-toolchain/registration-service/pull/474
- toolchain-e2e: https://github.com/codeready-toolchain/toolchain-e2e/pull/1054
- ksctl: https://github.com/kubesaw/ksctl/pull/84

### CODE REVIEW

**Review by xcoulon — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -128,3 +128,7 @@ require (
 )
 
 go 1.20
+
+replace github.com/codeready-toolchain/api => github.com/metlos/api v0.0.0-20240917153600-6b959ea7e5c1
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/metlos/toolchain-common v0.0.0-20240918131518-c886ba1e7ba0
```
Comment (alexeykazakov) [inline go.mod:134]: I thought we have discussed it multiple times and agreed not to include the dependency updates in the CRD update PRs :)

**Review by alexeykazakov — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (metlos) [reply go.mod]: removed

**Comment by metlos** [comment]
/retest

This seems environmental:

```
199. attempt (out of 200) of waiting for CRD memberoperatorconfigs.toolchain.dev.openshift.com to be available in the cluster and the InstallPlan to be complete
The installation takes suspiciously too long - deleting jobs. This may happen and is caused by flaky OLM installation.
```

**Comment by MatousJobanek** [comment]
@metlos this is not an environmental issue - it's failing that OLM cannot update ToolchainCluster CRD with the version. The problem is most likely the required `spec.apiEndpoint` field - it looks like that we need to make it optional first :man_facepalming: 
```
   status:
    bundleLookups:
    - catalogSourceRef:
        name: member-operator-c7c5a23
        namespace: toolchain-member-07140122
      identifier: toolchain-member-operator.v0.0.543-commit-c7c5a23
      path: quay.io/codeready-toolchain-test/member-operator-bundle:from.host-operator.PR1094.a749281
      properties: '{"properties":[{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"Workspace","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"MemberOperatorConfig","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"UserAccount","version":"v1alpha1"}},{"type":"olm.package","value":{"packageName":"toolchain-member-operator","version":"0.0.543-commit-c7c5a23"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"MemberStatus","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"ToolchainCluster","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"Idler","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"SpaceBindingRequest","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"NSTemplateSet","version":"v1alpha1"}},{"type":"olm.gvk","value":{"group":"toolchain.dev.openshift.com","kind":"SpaceRequest","version":"v1alpha1"}}]}'
      replaces: ""
    catalogSources:
    - member-operator-c7c5a23
    conditions:
    - lastTransitionTime: "2024-10-07T14:08:16Z"
      lastUpdateTime: "2024-10-07T14:08:16Z"
      message: 'error validating existing CRs against new CRD''s schema for "toolchainclusters.toolchain.dev.openshift.com":
        error validating toolchain.dev.openshift.com/v1alpha1, Kind=ToolchainCluster
        "toolchain-host-07140122/member-ci-op-qi3khbci-1de72.devsandboxci.devcluster.openshift.1":
        updated validation is too restrictive: [].spec.apiEndpoint: Required value'
      reason: InstallComponentFailed
      status: "False"
      type: Installed
    message: 'error validating existing CRs against new CRD''s schema for "toolchainclusters.toolchain.dev.openshift.com":
      error validating toolchain.dev.openshift.com/v1alpha1, Kind=ToolchainCluster
      "toolchain-host-07140122/member-ci-op-qi3khbci-1de72.devsandboxci.devcluster.openshift.1":
      updated validation is too restrictive: [].spec.apiEndpoint: Required value'
    phase: Failed 
```

**Pull request merged**